src/app/qgslabelinggui.cpp

```diff
@@ -611,6 +611,15 @@ void QgsLabelingGui::updateUi()
   {
     chkMergeLines->setToolTip( QString() );
   }
+
+  mCoordXDDBtn->updateFieldLists();
```

Comment:
Why are these calls to updateFieldLists() required? The field list is only used in the popup menu, and updateFieldList is already called before the menu is shown

Reply:
The field list is also used in updateGui to properly update the icon.